From c91c5ba776936e5532b18b54be02d69761f9ed35 Mon Sep 17 00:00:00 2001
From: Jayanthi A <jayanthi.annadurai@broadcom.com>
Date: Mon, 7 May 2012 11:04:13 +0530
Subject: [PATCH 1358/1532] XLP8xx NAE: Fixed interlaken interface statistics

Using hal function to read interface counters from cortina
interlaken chip.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/broadcom/nae/xlp_hw.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/broadcom/nae/xlp_hw.c b/drivers/net/ethernet/broadcom/nae/xlp_hw.c
index 8cfc051..70fe001 100755
--- a/drivers/net/ethernet/broadcom/nae/xlp_hw.c
+++ b/drivers/net/ethernet/broadcom/nae/xlp_hw.c
@@ -377,8 +377,10 @@ void xlp_get_mac_stats(struct net_device *dev, struct net_device_stats *stats)
 	unsigned long long val;
 #endif
 
-	if (priv->type == INTERLAKEN_IF)
-		return;
+        if (priv->type == INTERLAKEN_IF) {
+                nlm_hal_get_ilk_mac_stats(priv->node, priv->block, priv->phy.addr, stats);
+                return;
+        }
 
 	stats->tx_packets = nlm_hal_read_mac_reg(priv->node, priv->block, priv->index, TX_PACKET_COUNTER);
 #ifdef CONFIG_64BIT
-- 
1.9.1

